pkg/aws/ec2.go

```diff
@@ -134,26 +136,52 @@ func AssertEC2VolumeEncrypted(t *testing.T, ctx context.Context, client EC2Clien
 }
 
 // AssertVolumeType asserts the right volume type & associated throughput
-func AssertEC2VolumeType(t *testing.T, ctx context.Context, client EC2Client, input AssertEC2VolumeTypeInput) {
+func AssertEC2VolumeType(t *testing.T, ctx context.Context, client EC2Client, input AssertVolumeAttributesInput) {
 
 	instance, err := getEC2InstanceByInstanceIDE(ctx, client, input.InstanceID)
-	assert.Nil(t, err)
+	// assert.Nil(t, err)
+	require.NoError(t, err)
 
 	for _, v := range instance.BlockDeviceMappings {
-		if *v.DeviceName == input.DeviceID {
-			volume, err := getEC2VolumeByVolumeIDE(ctx, client, *v.Ebs.VolumeId)
-			assert.Nil(t, err)
-			assert.Equal(t, input.VolumeType, *volume.volumeType, "Volume with device ID '%s' for instance '%s' used the right volume type.", input.DeviceID, input.InstanceID)
-			if *volume.Encrypted == trueVal {
-				deviceEncrypted = true
-			}
-			if input.VolumeType != "gp2" {
-				assert.Equal(t, input.VolumeIops, *volume.iops, "Volume with device ID '%s' for instance '%s' used the right iops associated to volume.", input.DeviceID, input.InstanceID)
-				assert.Equal(t, input.VolumeThroughput, *volume.volumeThroughput, "Volume with device ID '%s' for instance '%s' used the right threshold associated to volume.", input.DeviceID, input.InstanceID)
-			}
+		volume, err := getEC2VolumeByVolumeIDE(ctx, client, *v.Ebs.VolumeId)
+		// assert.Nil(t, err)
+		require.NoError(t, err)
+		assert.Equal(t, input.VolumeType, *volume.VolumeType, "Volume with device ID '%s' used the right volume type.", input.DeviceID)
```

Comment:
**Suggestion:** Since this message is logged when the assertion _fails_, I think this verbiage is more appropriate.
```suggestion
		assert.Equal(t, input.VolumeType, *volume.VolumeType, "Volume with device ID '%s' does not have the right volume type.", input.DeviceID)
```